# 14.75 problem set 1

## Question 1

Create data frame with 100 observations with

a) A variable `t` running from `1` to `100` *bold text*

b) `alpha` which has value `3` for all 100 observations

c) Another variable `e_t`, a normal variable with mean 0 and standard deviation 1

d) `x_t` a random uniform variable over [0, 1]

1+1

My name is Max

In [26]:
1 + 1

[1] 2

In [11]:
SIZE = 100
set.seed(0)

In [12]:
df = data.frame(t=1:SIZE,
                alpha=3,
                e_t=rnorm(100),
                x_t=runif(100))

e) An outcome variable `y_t` where `beta=2`, `y_t` = `alpha + beta * x_t + e_t`.

In [13]:
BETA = 2
df$y_t = df$alpha + BETA * df$x_t + df$e_t

f) Estimate `beta_hat`

  i. Test H0: `beta = 0`
  ii. Test H0: `beta = 1.2`

In [25]:
summary(lm(y_t ~ x_t, data=df))


Call:
lm(formula = y_t ~ x_t, data = df)

Residuals:
     Min       1Q   Median       3Q      Max 
-2.23533 -0.60614 -0.06678  0.56760  2.50565 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept)   3.0983     0.1656  18.712  < 2e-16 ***
x_t           1.8257     0.3223   5.665 1.48e-07 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.8858 on 98 degrees of freedom
Multiple R-squared:  0.2467,	Adjusted R-squared:  0.239 
F-statistic: 32.09 on 1 and 98 DF,  p-value: 1.476e-07


In [15]:
confint(lm(y_t ~ x_t, data=df))

2.5 %    97.5 %  
(Intercept) 2.769741 3.426900
x_t         1.186110 2.465289

**95% CI [1.19, 2.47] does not include zero, so we can reject B=0, but it does include 1.2, so we cannot reject B=1.2.**

g) Create `v_t` as a random normal variable with mean 0. Generate `q_t` as `q_t = x_t + 2*x_t^3 + v_t`.

What is the correlation between q and x?

Generate outcome variable `z_t` where `beta=2`, `gamma=3`, `z_t = alpha + beta * x_t + gamma * q_t + e_t`.

Estimate `beta_hat` from (misspecified) model:
`z_t = alpha + beta * x_t + u_t`

Test H0: `beta = 2`.

Discuss.

In [16]:
df$v_t = rnorm(SIZE)

In [18]:
df$q_t = df$x_t + 2 * df$x_t ^ 3 + df$v_t

Correlation:

In [20]:
cor(df$q_t, df$x_t)

[1] 0.6420868

In [21]:
GAMMA = 3

In [22]:
df$z_t = df$alpha + BETA * df$x_t + GAMMA * df$q_t + df$e_t

In [23]:
summary(lm(z_t ~ x_t, data=df))


Call:
lm(formula = z_t ~ x_t, data = df)

Residuals:
    Min      1Q  Median      3Q     Max 
-8.6316 -2.4583  0.2057  2.1837  7.5694 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept)   1.6213     0.6186   2.621   0.0102 *  
x_t          11.4267     1.2041   9.490 1.55e-15 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 3.309 on 98 degrees of freedom
Multiple R-squared:  0.4789,	Adjusted R-squared:  0.4736 
F-statistic: 90.06 on 1 and 98 DF,  p-value: 1.548e-15


In [24]:
confint(lm(z_t ~ x_t, data=df))

2.5 %    97.5 %   
(Intercept) 0.393792  2.848848
x_t         9.037274 13.816108

**Estimated `beta` = 11.43. 95% CI = [9.04, 13.82] so we can reject `beta=2`.
This is the result of omitted variable bias: `z_t` is being affected by `q_t` which is correlated with `x_t` in a nonlinear way that `x_t` can't soak up alone (if it were linear we'd expect `beta_hat = beta + gamma = 5`, rather than 11.43).**

## Question 2

Regressions